Configurations/unix-Makefile.tmpl

```diff
@@ -857,7 +857,7 @@ EOF
 $target: $args{generator}->[0] $deps
 	( trap "rm -f \$@.*" INT 0; \\
 	  $generator \$@.S; \\
-	  \$(CC) $incs \$(CFLAGS) \$(CPPFLAGS) -E \$@.S | \\
+	  \$(CPP) $incs $cppflags \$@.S | \\
```

Comment:
This didn't work out. Trouble is that some assembly modules, non-x86[_64] ones, rely on pre-defines that are affected by platform flags. Platform flags such as -march, -mcpu, -m64. In other words one can't use CPPFLAGS alone, not when pre-processing assembly. Well, if we pre-processed C, it's not given that it would work out either. It's probably appropriate to say that pre-processing any **code** can require CFLAGS rather than CPPFLAGS alone.

Reply:
That isn't quite correct, isn't it?  *Preprocessing* any file will (shouls) do what the preprocessor does and nothing more, i.e. tokenize everything, process `#` directives, and expand macros.

*However*, it may be that some of those flags do add something that affects preprocessing, such as change certain built in macros.  Wouldn't that mean that such flags should really be considered cppflags?

Reply:
(BTW, I've already thought about that regarding the VMS `/POINTER_SIZE` qualifier, except it doesn't matter for the ia64 assembler code anyway...

Reply:
> That isn't quite correct, isn't it?

I'm sorry, that was wrong of me to say...  I misread what you wrote.

But, it seems that we are still talking about the same thing, but with two different solutions:

1.  reinstate `$(CFLAGS)` (or rather, the intention dependent variant) back into that command line, or
2.  move some flags to the `cppflags` attribute in configs, based on how they affect built in macros.

The 2nd solution has me wonder, are there flags that affect both built in macros and linking?  Since `$(CPPFLAGS)` aren't used with the latter, we might be looking at a bit of duplication, probably into the `lflags` attribute, or possibly `ex_libs`.

Reply:
> The 2nd solution has me wonder, are there flags that affect both built in macros and linking?

Yes. For example -m64 [and its equivalents] would customarily have to be specified at both compile and link. And yes, "compile" refers even to pre-processing of some [most?] non-x86_64 assembly. Well, some linkers can figure it out, that all object files are of same kind, but it can't be taken for universally granted.

Reply:
> 1 vs. 2

Do note that I said "it's *probably* appropriate to say ... CFLAGS". I meant that I tried not to make any specific suggestion. At least didn't actually intend to at that point. But now, after some consideration, I'd say whatever is passed when you compile .c. It's simpler and presumably more robust in long run.

[On related side[!] and more general note, as for classifying flags [and parameterizing things]. I'd argue that it's important to not overdo it, i.e. I'm not directly in favour of introducing infinite number of combinations and parameterizations. In other words I'd say it's as important to look for ways on how to reduce this kind of stuff, as to look for ways to introduce them. In yet other words what I'm trying to say is that I occasionally feel it's too much, and hence less obvious, and hence harder to grasp than necessary...]

Reply:
Ok, I addressed this in #5436